src/components/elements/tableFilters/filters/TableControlPopover.tsx

```diff
@@ -46,7 +51,8 @@ const TableControlPopover = (props: TableControlPopoverProps): JSX.Element => {
         {...bindTrigger(popupState)}
       >
         {label}
-        {filterCount > 0 && <> ({filterCount})</>}
+        {!isTiny && filterHint && <> ({filterHint})</>}
+        {filterCount > 0 && (isTiny || !filterHint) && <> ({filterCount})</>}
```

Comment:
This is fine (And we have plenty of logic like this in the codebase!) but it could be a bit more readable. I think a pattern like this makes it easier to add comments:

```
const filterSummary = useMemo(() => {
  if (filterCount === 0) return;
  // show num filters applied on mobile, or if we dont have a more descriptive hint
  if (isTiny || !filterHint) {
     return <> ({filterCount})</>
  }
  // show hint summarizing applied filters
  return <> ({filterHint})</>
},[]);



...
{label}
{filterSummary}
```